pkg/reconciler/revision/resources/deploy.go

```diff
@@ -118,6 +115,20 @@ var (
 	}
 )
 
+func addToken(tokenVolume *corev1.Volume, filename string, audience string, expiry *int64) {
+	if filename == "" || audience == "" || expiry == nil {
+		return
+	}
```

Comment:
These seem to indicate a programming error - worth returning it, or at least logging?

Reply:
Also, is `0` a valid expiry value, or should that be filtered, too?

Reply:
audience == "" is legal and should be silently skipped by design - this is normal behavior when the config parameter uses an empty string. 

filename and audience are equal in the current implementation and should be treated the same. 

I will remove the check for expiry - it is the work of an overly anxious coder